### ISSUE TITLE
How to enable Motion Chart in Highcharts angular?

### ISSUE THREAD

**Issue by mhdbouk** [highcharts/highcharts-angular#30]
Hello Highcharts team
I'm trying to work with motion to enable player inside my chart

I used to work with https://rawgit.com/larsac07/Motion-Highcharts-Plugin/master/motion.js

Before updating to the latest highcharts-angular, I was able to create a service that do the same thing that motion.js is doing. Bug after updating to the latest, my service is not working anymore, since `Highcharts.merge` / `Highcharts.splat` / `Highcharts.each` / `Highcharts.CreateElement` / `Highcharts.pick` are not working anymore.

getting the following error message

- Property 'merge' does not exist on type 'Static'.
- Property 'splat' does not exist on type 'Static'.
- Property 'each' does not exist on type 'Static'.
- Property 'CreateElement' does not exist on type 'Static'.
- Property 'pick' does not exist on type 'Static'.

Thank you for your help.

**Comment by KacperMadej** [comment]
Do you have any problems with other modules or plugins? Could you recreate the problem in a live demo or provide a way to recreate the problem?

You could try changing this line in the module (https://github.com/highcharts/highcharts-angular/commit/5673153457f606655e54e600a164d33d619165a4#diff-e5e546dd2eb0351f813d63d1b39dbc48R24) or load older version of the wrapper.

There's also v5-dev branch that might have code that could work better for you, so you could check that one too - https://github.com/highcharts/highcharts-angular/tree/v5-dev

**Comment by mhdbouk** [comment]
The problem is solved now
I added `motion.js` file to the solution
Then I set the `allowJs` to true inside `tsconfig.json`

```
"compilerOptions": {
    ...
    "allowJs": true,
    ...
}
```

adding the following line of code inside my component after the import of Highcharts
```
import * as Highcharts from 'highcharts';
require('path/to/motion.js')(Highcharts);
```

Thank you for your support 👍 🌹

**Comment by meghab1612** [comment]
Hi @mhdbouk ,

I am getting error Property 'merge' does not exist on type 'Static'.How did you solve that, which specific step solved it for you.

**Comment by mhdbouk** [comment]
Hello @meghab1612 
Did you add motion.js into the solution (asset folder) and allow js to be true in tsconfig.json?

When adding the motion.js file and you import motion into your component, it should work, when the error `'merge' does not exist on type 'Static'` that mean either motion is not included or not found

> The problem is solved now
> I added `motion.js` file to the solution
> Then I set the `allowJs` to true inside `tsconfig.json`
> 
> ```
> "compilerOptions": {
>     ...
>     "allowJs": true,
>     ...
> }
> ```
> adding the following line of code inside my component after the import of Highcharts
> 
> ```
> import * as Highcharts from 'highcharts';
> require('path/to/motion.js')(Highcharts);
> ```
> Thank you for your support 👍 🌹

**Comment by maddy25** [comment]
This didn't work for me. I tried adding motion.js file inside assets folder and turned allowJS to true.
import * as Highcharts from 'highcharts';
require('path/to/motion.js')(Highcharts);

Above code throws errror that Highcharts is not defined.

**Comment by KacperMadej** [comment]
Hi @maddy25 

First: if you could provide more info as mentioned (to another person) in one of my previous comments  https://github.com/highcharts/highcharts-angular/issues/30#issuecomment-388845296

Second: The plugin is not to be initialized with the `Highcharts` instance as an argument. The global variable `Highcharts` is expected by the plugin. You could set the `Highcharts` instance as such or call the `require` that will load the plugin in a controlled scope with the `Highcharts` instance set there - this might prevent some TS errors (it depends on your environment configuration if they will happen at all) and prevent from polluting the global scope of the original `require` call context.

tl;dr: The error is caused by motion plugin needing Highcharts as global variable.
